src/node/in_memory_ext.rs

```diff
@@ -219,22 +219,33 @@ impl<S: ForkSource + std::fmt::Debug + Clone + Send + Sync + 'static> InMemoryNo
     }
 
     pub fn mine_blocks(&self, num_blocks: Option<U64>, interval: Option<U64>) -> Result<bool> {
-        self.get_inner()
-            .write()
+        let num_blocks = num_blocks.unwrap_or_else(|| U64::from(1)).as_u64();
+        let interval_sec = interval.unwrap_or_else(|| U64::from(1)).as_u64();
+
+        if num_blocks == 0 {
+            return Err(anyhow!(
+                "Number of blocks must be greater than 0".to_string(),
+            ));
+        }
+
+        let bootloader_code = self
+            .get_inner()
+            .read()
             .map_err(|err| anyhow!("failed acquiring lock: {:?}", err))
-            .and_then(|mut writer| {
-                let num_blocks = num_blocks.unwrap_or_else(|| U64::from(1));
-                let interval_sec = interval.unwrap_or_else(|| U64::from(1));
-                if num_blocks.is_zero() {
-                    return Err(anyhow!(
-                        "Number of blocks must be greater than 0".to_string(),
-                    ));
-                }
-                utils::mine_empty_blocks(&mut writer, num_blocks.as_u64(), interval_sec.as_u64())?;
-                tracing::info!("👷 Mined {} blocks", num_blocks);
+            .map(|inner| inner.system_contracts.contracts_for_l2_call().clone())?;
+        for i in 0..num_blocks {
+            if i != 0 {
+                // Accounts for the default increment of 1 done in `seal_block`. Note that
+                // there is no guarantee that blocks produced by this method will have *exactly*
+                // `interval` seconds in-between of their respective timestamps. Instead, we treat
+                // it as the minimum amount of time that should have passed in-between of blocks.
+                self.time.increase_time(interval_sec - 1);
```

Comment:
What would happen if `interval_sec` is `Some(0)`?

Reply:
Good catch! Replaced `-` with `saturating_sub` which IMO is the correct behaviour here (the interval will still effectively be `1` but as the comment states we don't guarantee that it will be exactly `0` anyway).